### PR TITLE
Job e2e

### PR DESCRIPTION
**Pull request by soltysh**
Now that I have all the pieces in place I'm working entirely on e2e. This depends on #14046 and #14343.

### CODE REVIEW

**Comment by ncdc** [comment]
@kubernetes/rh-cluster-infra

**Comment by soltysh** [comment]
It covers:
* running a job
* running a failing job
* scaling job up and down
* stopping a job
Did I miss something? 
@deads2k fyi

**Comment by soltysh** [comment]
The e2e covers:
* running a job
* running a failing job
* scaling job up and down
* stopping a job 

Did I miss something? 

I'll rebase this once the aforementioned PRs will get merged.

@mikedanese @erictune ptal
@deads2k @sdminonne fyi

**Comment by lavalamp** [comment]
Which commits are we reviewing here?

**Comment by mikedanese** [comment]
@lavalamp last one. I don't mind doing this review.

**Comment by erictune** [comment]
Needs rebase after I merged Job update.

**Comment by erictune** [comment]
@soltysh

**Comment by erictune** [comment]
top-level file job.yaml not needed?

**Comment by erictune** [comment]
Needs to adjust for #14156 when rebasing.

File: job.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+kind: Job
```
Comment (erictune) [inline job.yaml]: delete this file or put it in a subdir.

**Comment by erictune** [comment]
working on a version of this that is rebased and has some improvements.

**Comment by erictune** [comment]
Superceded by #14566

**Pull request closed without merging**